dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/model/ApplicationModel.java

```diff
@@ -72,4 +72,12 @@ public static void initProviderModel(String serviceName, ProviderModel providerM
             LOGGER.warn("Already register the same:" + serviceName);
         }
     }
+
+    /**
+     * For unit test
+     */
+    public static void reset() {
+        providedServices.clear();
```

Comment:
I beleive providedServices and consumedServices would be never null here?

Reply:
Sorry my bad, it is initialize in this class while it is created. Plz ignore this comment 😄

Reply:
Just one thought, as you are indicating it is only for unit test, should we mark access specifier as protected. Which could help to use it more knowingly. This is just an thought.

Reply:
It cannot be protected, since it's used in telnet's unit test.

Reply:
ok.